diff -Nur inspircd.orig/conf/inspircd.conf.example.rej inspircd.1.1.2-m_no_op_on_channel_create/conf/inspircd.conf.example.rej
--- inspircd.orig/conf/inspircd.conf.example.rej	2009-04-26 14:42:57.000000000 -0600
+++ inspircd.1.1.2-m_no_op_on_channel_create/conf/inspircd.conf.example.rej	2009-04-26 14:42:57.000000000 -0600
@@ -1,11 +1,14 @@
---- inspircd.conf.example
-+++ inspircd.conf.example
-@@ -571,7 +571,7 @@
- # the default of 'inspircd.pid' is used.                              #
- #                                                                     #
+--- inspircd.conf.example	(revision 6664)
++++ inspircd.conf.example	(working copy)
+@@ -1431,6 +1431,11 @@
+ #<module name="m_nokicks.so">
  
--#<pid file="/path/to/inspircd.pid">
-+<pid file="/var/run/inspircd/inspircd.pid">
+ #-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#
++# No op on channel create module: Removes op privs from the modes an
++# user gets when creating a new channel.
++#<module name="m_no_op_on_channel_create.so">
++
++#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#
+ # No nicks module: Adds the +N channel mode
+ #<module name="m_nonicks.so">
  
- #-#-#-#-#-#-#-#-#-#-#-#-#- BANLIST LIMITS #-#-#-#-#-#-#-#-#-#-#-#-#-#-#
- #                                                                     #
diff -Nur inspircd.orig/conf/modules.conf.example inspircd.1.1.2-m_no_op_on_channel_create/conf/modules.conf.example
--- inspircd.orig/conf/modules.conf.example	2009-03-06 03:04:43.000000000 -0700
+++ inspircd.1.1.2-m_no_op_on_channel_create/conf/modules.conf.example	2009-04-26 14:44:40.000000000 -0600
@@ -1054,6 +1054,11 @@
 #<module name="m_nokicks.so">
 
 #-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#
+# No op on channel create module: Removes op privs from the modes an
+# user gets when creating a new channel.
+#<module name="m_no_op_on_channel_create.so">
+
+#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#-#
 # No nicks module: Adds the +N channel mode, as well as the +b N:
 # extended bantype. +N stops all users from changing their nick,
 # the +b N: extban stops anyone from matching a +b N:nick!user@host
diff -Nur inspircd.orig/modules/m_no_op_on_channel_create.cpp inspircd.1.1.2-m_no_op_on_channel_create/modules/m_no_op_on_channel_create.cpp
--- inspircd.orig/modules/m_no_op_on_channel_create.cpp	1969-12-31 17:00:00.000000000 -0700
+++ inspircd.1.1.2-m_no_op_on_channel_create/modules/m_no_op_on_channel_create.cpp	2009-04-26 14:42:57.000000000 -0600
@@ -0,0 +1,75 @@
+/*       +------------------------------------+
+ *       | Inspire Internet Relay Chat Daemon |
+ *       +------------------------------------+
+ *
+ *  InspIRCd: (C) 2002-2007 InspIRCd Development Team
+ * See: http://www.inspircd.org/wiki/index.php/Credits
+ *
+ * This program is free but copyrighted software; see
+ *            the file COPYING for details.
+ *
+ * ---------------------------------------------------
+ */
+
+#include "users.h"
+#include "channels.h"
+#include "modules.h"
+#include "hashcomp.h"
+#include "inspircd.h"
+#include "wildcard.h"
+
+/* $ModDesc: Disables op on channel creation */
+
+class ModuleOpOnCreateChannel : public Module
+{
+ public:
+	ModuleOpOnCreateChannel(InspIRCd* Me) : Module::Module(Me)
+	{
+	}
+	virtual Version GetVersion()
+	{
+		return Version(1,0,0,0,VF_VENDOR,API_VERSION);
+	}
+
+	void Implements(char* List)
+	{
+		List[I_OnUserPreJoin] = 1;
+	}
+
+	virtual int OnUserPreJoin(userrec* user, chanrec* chan, const char* cname, std::string &privs)
+	{
+		string::size_type loc = privs.find('@');
+		if (loc != string::npos)
+		{
+			privs.erase(loc);
+		}
+		return 0;
+	}
+};
+
+// stuff down here is the module-factory stuff. For basic modules you can ignore this.
+
+class ModuleOpOnCreateChannelFactory : public ModuleFactory
+{
+ public:
+	ModuleOpOnCreateChannelFactory()
+	{
+	}
+	
+	~ModuleOpOnCreateChannelFactory()
+	{
+	}
+	
+	virtual Module * CreateModule(InspIRCd* Me)
+	{
+		return new ModuleOpOnCreateChannel(Me);
+	}
+	
+};
+
+
+extern "C" void * init_module( void )
+{
+	return new ModuleOpOnCreateChannelFactory;
+}
+
